compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2462,8 +2462,16 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                 self.normalize_ty(ast_ty.span, array_ty)
             }
             hir::TyKind::Typeof(ref e) => {
-                tcx.sess.emit_err(TypeofReservedKeywordUsed { span: ast_ty.span });
-                tcx.type_of(tcx.hir().local_def_id(e.hir_id))
+                let ty = tcx.type_of(tcx.hir().local_def_id(e.hir_id));
```

Comment:
does this report a cycle error if we do something like:

```rust
fn foo() -> typeof(foo) {}
```

Reply:
it does

Reply:
hrm.. wonder if that's ez to fix... probably not. should be ok, since this is illegal syntax anyways.

Reply:
Does this ICE? If then it should be fixed. Not necessary in this PR though.

Reply:
It doesn't ICE, it just shows the usual cycle error like "cycle detected when computing function signature of `foo`"
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c657d61f3db8f63bce43cffa68c1f85b